test/res.append.js

```diff
@@ -89,16 +91,26 @@ describe('res', function () {
       })
 
       app.use(function (req, res) {
-        res.append('Set-Cookie', 'bar=baz')
+        res.append('Set-Cookie', 'fizz=buzz')
         res.end()
       })
 
       request(app)
-      .get('/')
-      .expect(function (res) {
-        should(res.headers['set-cookie']).eql(['foo=bar; Path=/', 'bar=baz'])
-      })
-      .expect(200, done)
+        .get('/')
+        .expect(200)
+        .expect(shouldHaveHeaderValues('Set-Cookie', ['foo=bar; Path=/', 'fizz=buzz']))
+        .end(done)
     })
   })
 })
+
+function shouldHaveHeaderValues (key, values) {
+  return function (res) {
+    var headers = res.headers[key.toLowerCase()]
+    assert.ok(headers, 'should have header "' + key + '"')
+    assert.strictEqual(headers.length, values.length, 'should have ' + values.length + ' occurances of "' + key + '"')
+    for (var i = 0; i < values.length; i++) {
+      assert.strictEqual(headers[i], values[i])
+    }
+  }
+}
```

Comment:
**Issue**: The function 'shouldHaveHeaderValues' does not handle the case where the header does not exist or has a different case, which could lead to runtime errors if the header is not found or the case does not match. <br> **Fix**: Modify the function to handle cases where the header might not exist or might have a different case by ensuring the header key is converted to lower case and checking if the header actually exists before proceeding with further checks. <br> **Code Suggestion**: 
 ```
 @@ -109,7 +109,11 @@
      var headers = res.headers[key.toLowerCase()]
 -    assert.ok(headers, 'should have header \"' + key + '\"')
 +    if (!headers) {
 +      throw new Error('Header ' + key + ' not found.');
 +    }
      assert.strictEqual(headers.length, values.length, 'should have ' + values.length + ' occurances of \"' + key + '\"')
      for (var i = 0; i < values.length; i++) {
        assert.strictEqual(headers[i], values[i])
      }
 ```